### PR TITLE
Database Versioning Completed

### PR DESCRIPTION
**Pull request by mananwason**
@creativepsyco Please check and merge

### CODE REVIEW

**Comment by mananwason** [comment]
#7

**Comment by creativepsyco** [comment]
@mananwason you should assign me  the PR, the option is on the right, otherwise I might ignore it coz of email spam.

![image][1]

[1]: https://www.dropbox.com/s/yo4u1xp3eap7wfo/Screenshot%202015-06-15%2013.31.09.png?raw=1

**Comment by mananwason** [comment]
Really sorry @creativepsyco. I forgot assigning you this time

File: app/src/main/java/org/fossasia/openevent/dbutils/DbContract.java [hunk 1]
```diff
@@ -41,15 +41,14 @@ public DbContract() {
         public static final String CREATE_TABLE =
                 "CREATE TABLE " + TABLE_NAME
                         + " ("
-                        + _ID + INT_TYPE + PRIMARY_KEY + COMMA_SEP
-                        + EVENT_ID + INT_TYPE + COMMA_SEP
+                        + VER_ID + INT_TYPE + PRIMARY_KEY + COMMA_SEP
+                        //+ EVENT_ID + INT_TYPE + COMMA_SEP
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/dbutils/DbContract.java]: Try not to change the schema a lot in the commits.

File: app/src/main/java/org/fossasia/openevent/dbutils/DbSingleton.java [hunk 2]
```diff
@@ -170,6 +171,42 @@ public Session getSessionById(int id) throws ParseException {
...
+                    cursor.getInt(cursor.getColumnIndex(DbContract.Versions.VER_ID)),
+                    cursor.getInt(cursor.getColumnIndex(DbContract.Versions.VER_EVENT)),
+                    cursor.getInt(cursor.getColumnIndex(DbContract.Versions.VER_TRACKS)),
+                    cursor.getInt(cursor.getColumnIndex(DbContract.Versions.VER_SESSIONS)),
+                    cursor.getInt(cursor.getColumnIndex(DbContract.Versions.VER_SPONSORS)),
+                    cursor.getInt(cursor.getColumnIndex(DbContract.Versions.VER_SPEAKERS)),
+                    cursor.getInt(cursor.getColumnIndex(DbContract.Versions.VER_MICROLOCATIONS))
+            );
+            cursor.close();
+            return current_version;
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/dbutils/DbSingleton.java]: naming convention no underscores.

File: app/src/main/java/org/fossasia/openevent/dbutils/DbContract.java (hunk 1 above)
Comment (mananwason) [inline app/src/main/java/org/fossasia/openevent/dbutils/DbContract.java]: Cool I guess this is pretty much finalised schema until some changes are made in server.

**Comment by mananwason** [comment]
@creativepsyco please check.

**Comment by creativepsyco** [comment]
@mananwason Please write tests. There is no use of coding if there aren't any tests. DB level tests are a must.

**Comment by mananwason** [comment]
@creativepsyco Once I'm done with the UI I'm doing right now, i'll start writing tests

**Comment by creativepsyco** [comment]
I would suggest writing tests at the DB layer too, just in case, it doesn't require any UI for that

**Comment by mananwason** [comment]
Good point. I've already started writing some tests on database.

**Pull request merged**